octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/entitydetail/field/DescriptionComposite.java

```diff
@@ -1,69 +1,160 @@
+                    Util.getUiDataFromModel(entityModel.getValue((EntityFieldsConstants.FIELD_DESCRIPTION))));
+        } catch (IOException e) {
+        }
+
+        String descriptionText = "<html><body bgcolor =" + getRgbString(backgroundColor) + ">" + "<font color ="
+                + getRgbString(foregroundColor) + ">"
+                + descriptionFromServerRemodeled
+                + "</font></body></html>";
+        if (descriptionText.equals("<html><body bgcolor =" + getRgbString(backgroundColor) + ">" + "<font color ="
+                + getRgbString(foregroundColor) + ">" + "</font></body></html>")) {
+
+            return "<html><body bgcolor =" + getRgbString(backgroundColor) + ">" + "<font color ="
+                    + getRgbString(foregroundColor) + ">" + "No description" + "</font></body></html>";
+        } else {
+            return descriptionText;
+        }
+
+    }
+
+    private static String getRgbString(Color color) {
+        return "rgb(" + color.getRed() + "," + color.getGreen() + "," + color.getBlue() + ")";
+    }
+
+    private String downloadPictures(String descriptionField) throws IOException {
+        String tmpPath = System.getProperty("java.io.tmpdir");
+        String baseUrl = Activator.getConnectionSettings().getBaseUrl();
+        File tmpDir = new File(tmpPath);
+        octanePhotosDir = new File(tmpDir, "Octane_pictures");
+
+        if (!new File(tmpDir, "Octane_pictures").exists()) {
+            octanePhotosDir.mkdir();
+        }
+        String octanePhotosPath = octanePhotosDir.getAbsolutePath();
+
+        Document descriptionParser = Jsoup.parse(descriptionField);
+        Elements link = descriptionParser.getElementsByTag("img");
+
+        for (Element el : link) {
+            String pictureLink = el.attr("src");
+            if (pictureLink.contains("/api/shared_spaces")) {
+                el.attr("src", baseUrl + pictureLink);
+                pictureLink = el.attr("src");
+            }
+
+            if (!pictureLink.contains(Activator.getConnectionSettings().getBaseUrl())) {
+                continue;
+            }
+            int index = pictureLink.lastIndexOf("/");
+            String pictureName = pictureLink.substring(index + 1, pictureLink.length());
+            String picturePath = octanePhotosPath + "\\" + pictureName;
+            if (!new File(octanePhotosDir, pictureName).exists()) {
+                saveImage(pictureLink, picturePath);
+            }
+            ;
+            el.attr("src", picturePath);
+        }
+        return descriptionParser.toString();
+    }
+
+    private void saveImage(String pictureLink, String octanePhotosName) throws IOException {
+
+        if (isUserLoggedIn() == true) {
+            HttpResponse httpResponse = ClientLoginCookie.getDataForImage(pictureLink);
+            InputStream is = httpResponse.getContent();
+
+            OutputStream os = new FileOutputStream(octanePhotosName);
+            IOUtils.copy(is, os);
```

Comment:
@vladbenchea We should use "try/finally" to make sure that if an exception occurs, the streams are correctly closed

Reply:
I will correct this. Thanks!